### ISSUE TITLE
 Yolov5 classification CHW to BCWH

### ISSUE THREAD

**Issue by xellDart** [Tencent/ncnn#3724]
Hi , I have one question, I train my dataset using classifier branch on yolov5 (https://github.com/ultralytics/yolov5/blob/classifier/classifier.py), with accuracy of 0.981, later i use this code using ncnn for classification task:

The problem I have is the processing of the image and I don't understand how to apply this code to c++:
https://gist.github.com/xellDart/3b77da5152dbc1ada9008ba6be68fa1d

current implementation on c++ (not work, recognize all images with same class)
https://gist.github.com/xellDart/e8760c6d7be1dd2a7090ea736dfc1d69

My big problem is translate numpy code to c++, for example, conversion HWC to CHW

```
im = np.ascontiguousarray(np.asarray(im).transpose((2, 0, 1))) # shape = [3,320,320]
```

For this I apply transpose layer

```
void transpose(const ncnn::Mat& in, ncnn::Mat& out)
{
    ncnn::Option opt;
    opt.num_threads = 2;
    opt.use_fp16_storage = false;
    opt.use_packing_layout = true;

    ncnn::Layer* op = ncnn::create_layer("Permute");

    // set param
    ncnn::ParamDict pd;
    pd.set(1, 5);// order_type

    op->load_param(pd);

    op->create_pipeline(opt);

    ncnn::Mat in_packed = in;
    {
        // resolve dst_elempack
        int dims = in.dims;
        int elemcount = 0;
        if (dims == 1) elemcount = in.elempack * in.w;
        if (dims == 2) elemcount = in.elempack * in.h;
        if (dims == 3) elemcount = in.elempack * in.c;

        int dst_elempack = 1;
        if (op->support_packing)
        {
            if (elemcount % 8 == 0 && (ncnn::cpu_support_x86_avx2() || ncnn::cpu_support_x86_avx()))
                dst_elempack = 8;
            else if (elemcount % 4 == 0)
                dst_elempack = 4;
        }

        if (in.elempack != dst_elempack)
        {
            convert_packing(in, in_packed, dst_elempack, opt);
        }
    }

    // forward
    op->forward(in_packed, out, opt);

    op->destroy_pipeline(opt);

    delete op;
}
```

I use  pd.set(1, 5);// order_type based on  https://github.com/Tencent/ncnn/blob/master/src/layer/permute.cpp

```
 // order_type
        // 0 = w h c
        // 1 = h w c
        // 2 = w c h
        // 3 = c w h
        // 4 = h c w
        // 5 = c h w
```

Later I need to convert Tensor to BCWH

```
  im = torch.tensor(im).float().unsqueeze(0) / 255.0 # shape = [1, 3,320,320]
```

But I have no idea how do this...

My last question is about normalization params, In python code I use 

```
normalize = lambda x, mean=0.5, std=0.25: (x - mean) / std
```

What would be the equivalent in c++ using ncnn?

Complete preprocess image code

```
 model = torch.jit.load('runs/train/exp/weights/best.torchscript')
    resize = torch.nn.Upsample(size=(size, size), mode='bilinear', align_corners=False)  # image resize

    # Image
    for filename in os.listdir(folder):
        filename = folder + filename
        im = Image.open(filename)  # HWC, BGR to RGB
        im = np.ascontiguousarray(np.asarray(im).transpose((2, 0, 1)))  # HWC to CHW
        im = torch.tensor(im).float().unsqueeze(0) / 255.0  # to Tensor, to BCWH, rescale
        im = resize(normalize(im))

        # Inference
        results = model(im)
        p = F.softmax(results, dim=1)  # probabilities
        i = p.argmax()  # max index
        print(f'{filename} prediction: {i} ({p[0, i]:.2f})')
```

Thanks

**Comment by ITCZhuxy** [comment]
Hello, is the Yolov5 Classification code that you wrote? Could you share it with me? I couldn't find an example of Yolov5 Classification.
